drivers/network/afd/afd/main.c

```diff
@@ -847,6 +848,52 @@ AfdDisconnect(PDEVICE_OBJECT DeviceObject, PIRP Irp,
     return UnlockAndMaybeComplete( FCB, Status, Irp, 0 );
 }
 
+NTSTATUS
+AfdQueryFsDeviceInfo(PDEVICE_OBJECT DeviceObject, PFILE_FS_DEVICE_INFORMATION Buffer, PULONG Length)
+{
+    if (*Length >= sizeof(FILE_FS_DEVICE_INFORMATION))
+    {
+        Buffer->Characteristics = 0;
+        Buffer->DeviceType = FILE_DEVICE_NAMED_PIPE;
+
+        *Length -= sizeof(FILE_FS_DEVICE_INFORMATION);
+
+        return STATUS_SUCCESS;
+    }
+    else
+    {
+        return STATUS_INVALID_PARAMETER;
```

Comment:
STATUS_BUFFER_TOO_SMALL or STATUS_INFO_LENGTH_MISMATCH would be more specific.

Reply:
Any preference between those 2?

Reply:
Ah.
I did not see this check in other places, but because I wasn't sure, I just added the check to be on the safe side.